deploy/lib/deployFunctions.js

```diff
@@ -5,50 +5,48 @@ const BbPromise = require("bluebird");
 module.exports = {
   deployFunctions() {
     this.serverless.cli.log("Deploying Functions...");
-    return BbPromise.bind(this)
-      .then(this.deployEachFunction)
-      .then(() =>
-        this.serverless.cli.log(
-          "Waiting for function deployments, this may take multiple minutes..."
-        )
-      )
-      .then(this.printFunctionInformationAfterDeployment);
+    return BbPromise.bind(this).then(this.deployEachFunction);
   },
 
   deployEachFunction() {
-    const promises = this.functions.map((func) =>
-      this.deployFunction(func.id, {})
+    return BbPromise.map(
+      this.functions,
+      (func) => {
+        return this.deployFunction(func.id, {})
+          .then((func) => {
+            this.serverless.cli.log(`Deploying ${func.name}...`);
+            return func;
+          })
+          .then((func) => this.waitForFunctionStatus(func.id, "ready"))
+          .then((func) => this.printFunctionInformationAfterDeployment(func))
+          .then((func) => this.waitForDomainsDeployment(func));
+      },
+      { concurrency: 5 }
```

Comment:
For reviewers: this is the most important change: allow to deploy functions 5 at a time (and wait until function is deployed completely before trying to deploy a new one).